src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+#include <OpenMS/CHEMISTRY/EnzymesDB.h>
+
+#include <QtCore/QFile>
+#include <QtCore/QProcess>
+#include <QDir>
+#include <QDebug>
+#include <iostream>
+#include <fstream>
+
+using namespace OpenMS;         
+using namespace std;             
+
+//-------------------------------------------------------------
+//Doxygen docu
+//-------------------------------------------------------------
+
+/**
+    @page TOPP_COMETAdapter COMETAdapter
+
+    @brief Identifies peptides in MS/MS spectra via COMET.
+
+<CENTER>
+    <table>
+        <tr>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> pot. predecessor tools </td>
+            <td VALIGN="middle" ROWSPAN=2> \f$ \longrightarrow \f$ COMETAdapter \f$ \longrightarrow \f$</td>
+            <td ALIGN = "center" BGCOLOR="#EBEBEB"> pot. successor tools </td>
+        </tr>
+        <tr>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> any signal-/preprocessing tool @n (in mzML format)</td>
+            <td VALIGN="middle" ALIGN = "center" ROWSPAN=1> @ref TOPP_IDFilter or @n any protein/peptide processing tool</td>
+        </tr>
+    </table>
+</CENTER>
+
+    @em COMET must be installed before this wrapper can be used. This wrapper
+    has been successfully tested with several versions of COMET.
+
+    This adapter supports relative database filenames, which (when not found in the current working directory) is looked up in
+    the directories specified by 'OpenMS.ini:id_db_dir' (see @subpage TOPP_advanced).
+
+    COMET settings not exposed by this adapter can be directly adjusted using an txt param file, which can be generated using comet -p.
+    By default, All (!) parameters available explicitly via this param file will take precedence over the wrapper parameters.
+
+    Parameter names have been changed to match names found in XTandem or MSGFPlus Search Engines, however some are Comet specific.
+    For a detailed description of all available parameters check the Comet documentation at http://comet-ms.sourceforge.net/parameters/parameters_201601/
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude TOPP_COMETAdapter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude TOPP_COMETAdapter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+
+class TOPPCOMETAdapter :
+  public TOPPBase
+{
+public:
+  TOPPCOMETAdapter() :
+    TOPPBase("COMETAdapter", "Annotates MS/MS spectra using COMET.")
+  {
+  }
+
+protected:
+  void registerOptionsAndFlags_()
+  {
+
+    registerInputFile_("in", "<file>", "", "Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerOutputFile_("out", "<file>", "", "Output file");
+    setValidFormats_("out", ListUtils::create<String>("idXML"));
+    registerInputFile_("database", "<file>", "", "FASTA file or pro file. Non-existing relative file-names are looked up via'OpenMS.ini:id_db_dir'", true, false, ListUtils::create<String>("skipexists"));
```

Comment:
in any case. either remove support for pro files or add it to the valid file types in the next line.